controllers/component_controller.go

```diff
@@ -293,8 +301,145 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Build.ContainerImage == "" {
+		component.Spec.Build.ContainerImage = r.ImageRepository + ":" + component.Namespace + "-" + component.Name
+	}
+
+	_, err = r.generateBuild(ctx, &component)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.Client.Status().Update(ctx, &component)
+	if err != nil {
+		log.Error(err, "Unable to update Component's status")
+	}
+
 	log.Info(fmt.Sprintf("Finished reconcile loop for %v", req.NamespacedName))
-	return ctrl.Result{}, nil
+	return ctrl.Result{}, err
+}
+
+func (r *ComponentReconciler) generateBuild(ctx context.Context, component *appstudiov1alpha1.Component) (ctrl.Result, error) {
```

Comment:
If we're not using `ctrl.Result` as a return value, can we just drop it from the function header?